src/sentry/services/hybrid_cloud/organization_mapping/impl.py

```diff
@@ -0,0 +1,75 @@
+from dataclasses import fields
+from typing import Any, Optional
+
+from django.db import IntegrityError, transaction
+
+from sentry.models.organizationmapping import OrganizationMapping
+from sentry.models.user import User
+from sentry.services.hybrid_cloud.organization_mapping import (
+    APIOrganizationMapping,
+    OrganizationMappingService,
+)
+
+
+class DatabaseBackedOrganizationMappingService(OrganizationMappingService):
+    def create(
+        self,
+        *,
+        user: User,
+        organization_id: int,
+        slug: str,
+        region_name: str,
+        idempotency_key: Optional[str] = "",
+        # There's only a customer_id when updating an org slug
+        customer_id: Optional[str] = None,
+    ) -> APIOrganizationMapping:
+        try:
+            with transaction.atomic():
+                # Creating an identical mapping should succeed, even if a record already exists
+                # with this slug. We allow this IFF the idempotency key is identical.
```

Comment:
This makes sense, I'd prefer if document string was also clear about this.

Reply:
@corps Updated